Doc/#444 - Taipy CLI documentation (#456)
Avaiga/taipy-doc: docs/manuals/core/versioning/configuration.md

> This section describes how to configure the Taipy version management system and explains different configuring methods, with the command-line interface (CLI) options or with a TOML file.

Reviewer: ```suggestion
Taipy uses the Config mechanism (Python or TOML) to configure the version management. A dedicated `Config` section named `CORE` can be used. Its attributes can be dynamically overwritten at runtime with command-line interface (CLI) options.

This section describes the various methods to configure the Taipy version management system.
```

  in *experiment* mode and only considers the entities attached to the version used.
  All other entities attached to different versions are filtered out.
  When the version is provided as a command line argument, a new *experiment* version
  is created using the version name provided. If no version is provided, a random string
  is used.
  If the version provided already exists, Taipy runs the application using the existing
  version only if the current configuration has not changed compared to the existing
  version.

- **--production [VERSION]**: With the `--production` argument, Taipy runs the application
  in *production* mode with the version provided. All existing entities are accessible.
  If the version provided already exists as an *experiment* version, it is converted
  to a production version.

- **--taipy-force**: With the `--taipy-force` argument, Taipy overrides a version even if
  the configuration has changed and run the application. Default to False.

- **--clean-entities**: With the `--clean-entities` argument, running a Taipy
  Core application cleans all current version entities before running the application.
  Default to False.

> # Configure using explicit TOML file
> 

Reviewer: What about this structure? So we also document the Python API.

```suggestion
# Configure using Taipy Config

You can also use the Taipy configuration mechanism. 

## Using Python

TODO document the Python API

## Using a TOML File

```

Author: I structured it as
```
# Configure using Taipy Config
# Configure using a TOML file
# Configure using the CLI
```
since there are 3 layers of configuration.
Is it ok?

Reviewer: sounds good to me

> You also provide a TOML file configuration with a specific section named [core].

Reviewer: We have the convention to name sections with an upper case. I am changing it in the code as well. for the 3.0 we will need to change the gui section as well.

```suggestion
You also provide a TOML file configuration with a specific section named [CORE].
```

Author: it is actually "core" lowercase at the moment.

Reviewer: Yep. I just created a PR to rename it.

Here is an example of a TOML file that set the mode to experiment, and force overriding the previous experiment version when there is conflicting Configuration.

```toml linenums="1" title="config.toml"

> [core]

Reviewer: ```suggestion
[CORE]
```

Author: it is actually "core" lowercase at the moment.

Reviewer: Yep. I just created a PR to rename it.
